scheds/rust/scx_mitosis/src/main.rs

```
@@ -125,31 +131,200 @@ impl<'a> Scheduler<'a> {
+            if cell_queue_decisions == 0 {
+                continue;
+            }
+
+            let mut queue_counts = [0; N];
+            for (i, &stat) in queue_stats_idx.iter().enumerate() {
+                queue_counts[i] = self.cell_stats_delta[cell][stat as usize];
             }
+
+            const MIN_CELL_WIDTH: usize = 2;
+            let cell_width: usize = max(MIN_CELL_WIDTH, (MAX_CELLS as f64).log10().ceil() as usize);
+
+            let prefix = format!("Cell {:width$} Decisions: ", cell, width = cell_width);
+
+            // Sum affinity violations for this cell
+            let scope_affn_viols: u64 = self.cell_stats_delta[cell][bpf_intf::cell_stat_idx_CSTAT_AFFN_VIOL as usize];
+
+            self.calculate_distribution_and_log(&queue_counts,
+                                                global_queue_decisions,
+                                                cell_queue_decisions,
+                                                scope_affn_viols,
+                                                &prefix)?;
+
         }
+        Ok(())
+    }
+
+    fn log_all_queue_stats(&self) -> Result<()> {
+        // The subset of cstats we care about.
+        // Local + Default + Hi + Lo = Total Decisions
+        // Affinity violations are not queue decisions, but
+        // will be calculated separately and reported as a percent of the total
+        let queue_stats_idx = [
+            bpf_intf::cell_stat_idx_CSTAT_LOCAL,
+            bpf_intf::cell_stat_idx_CSTAT_DEFAULT_Q,
+            bpf_intf::cell_stat_idx_CSTAT_HI_FALLBACK_Q,
+            bpf_intf::cell_stat_idx_CSTAT_LO_FALLBACK_Q,
+        ];
+        // Get total decisions
+        let global_queue_decisions: u64 = self.cell_stats_delta
+            .iter()
+            .flat_map(|cell| queue_stats_idx.iter().map(|&idx| cell[idx as usize]))
+            .sum();
+
+        // We don't want to divide by zero later, but this is never expected.
```

> never expected

Should we return an error, then?

---

Switched this to returning an error.